spharpy/spherical.py

```diff
@@ -1104,3 +1107,397 @@ def sph_identity_matrix(n_max, type='n-nm'):
+        if value != self._condon_shortley:
+            self._reset_compute_attributes()
+        self._condon_shortley = value
+
+    @property
+    def n_max(self):
+        """Get or set the spherical harmonic order."""
+        return self._n_max
+
+    @n_max.setter
+    def n_max(self, value):
+        """Get or set the spherical harmonic order."""
+        if value < 0:
+            raise ValueError("n_max must be a positive integer")
+        if value % 1 != 0:
+            raise ValueError("n_max must be an integer value")
+        if self.channel_convention == "fuma" and value > 3:
+            raise ValueError("n_max > 3 is not allowed with 'fuma' "
+                             "channel convention")
+        if self.normalization == "maxN" and value > 3:
+            raise ValueError("n_max > 3 is not allowed with 'maxN' "
+                             "normalization")
+        if int(value) != self._n_max:
+            self._reset_compute_attributes()
+            self._n_max = int(value)  # Cast to int for safety
+
+    @property
+    def coordinates(self):
+        """Get or set the coordinates object."""
+        return self._coordinates
+
+    @coordinates.setter
+    def coordinates(self, value):
+        """Get or set the coordinates object."""
+        if not isinstance(value, pf.Coordinates):
+            raise TypeError("coordinates must be a pyfar.Coordinates "
+                            "object or spharpy.SamplingSphere object")
+        if value.cdim != 1:
+            raise ValueError("Coordinates must be a 1D array")
+        if value.csize == 0:
+            raise ValueError("Coordinates cannot be empty")
+        if value != self._coordinates:
+            self._reset_compute_attributes()
+            self._coordinates = value
+
+    @property
+    def basis_type(self):
+        """Get or set the type of spherical harmonic basis."""
+        return self._basis_type
+
+    @basis_type.setter
+    def basis_type(self, value):
+        """Get or set the type of spherical harmonic basis."""
+        if value not in ["complex", "real"]:
+            raise ValueError("Invalid basis type, only "
+                             "'complex' and 'real' are supported")
+        if value != self._basis_type:
+            self._reset_compute_attributes()
+            self._basis_type = value
+
+    @property
+    def inverse_method(self):
+        """Get or set the type of inverse transform."""
+        return self._inverse_method
+
+    @inverse_method.setter
+    def inverse_method(self, value):
+        """Get or set the inverse transform type."""
+        # If the user passes "auto", require SamplingSphere and resolve it
+        if isinstance(value, str) and value == "auto":
+            if not isinstance(self.coordinates, SamplingSphere):
+                raise ValueError("'auto' is only valid if `coordinates` is a SamplingSphere.")
+            if self.coordinates.quadrature:
```

Comment:
I think this will raise an error for Coordinates objects because they dont have the property. This should fix it

```suggestion
            if type(coordinates) is SamplingSohere and self.coordinates.quadrature:
```

Reply:
done